src/examples/algorithms/Anomaly.cpp

```diff
@@ -0,0 +1,199 @@
+#include <iostream>     
+#include <fstream>      
+#include <vector>
+#include <string>
+#include <algorithm>    
+#include <iterator>     
+#include <boost/tokenizer.hpp>
+
+#include "nupic/encoders/ScalarEncoder.hpp"
+#include "nupic/algorithms/SpatialPooler.hpp"
+#include "nupic/algorithms/Cells4.hpp"
```

Comment:
let's work with `TemporalMemory` instead, as the API is cleaner and closer to the reference python impl. 
(I'd still be curious how to get the proper data from TP implementation)